### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by jerremyng**
Add navigation side bar as well as documentation content.

### CODE REVIEW

**Comment by cyqjoseph** [comment]
Nice update in our documentation, users should be more clear now in using our product, as well as potential developers who want to build on Dook!

**Comment by gauravuj** [comment]
that's a nice idea - makes it even better than other UGs where one has to scroll to the top

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -2,17 +2,11 @@
 layout: page
 title: Developer Guide
 ---
+Welcome to our Developer Guide! This resource is designed to provide developers with a overview of Dook's architecture. Whether you're new to the project or looking to extend its functionality, this guide is the best place to start learning how to contribute to Dook.
```
Comment (cyqjoseph) [inline docs/DeveloperGuide.md:5]: Great add!

File: docs/_sass/minima/_layout.scss [hunk 2]
```diff
@@ -207,6 +207,19 @@
   }
 }
 
+#markdown-toc {
```
Comment (cyqjoseph) [inline docs/_sass/minima/_layout.scss:210]: Nice styling!

**Pull request merged**